### ISSUE TITLE
The `packager.packList(..)` method doesn't work

### ISSUE THREAD

**Issue by 4ybakut2004** [skjolber/3d-bin-container-packing#440]
Hi!

There are 2 containers and 10 boxes. All boxes are placed in one container.

If I take 2 boxes of each type, then the `packager.packList(..)` method returns `null`, but I expect two containers to be enough. Even if I take 100 containers, the method returns `null`, although the number of boxes is only doubled.

Here is the source code and the console output:

```kotlin
import com.github.skjolber.packing.api.Box
import com.github.skjolber.packing.api.StackableItem
import com.github.skjolber.packing.packer.laff.LargestAreaFitFirstPackager
import com.github.skjolber.packing.api.Container

fun main(args: Array<String>) {
    Test(boxCountPerStackableItem = 1, numberOfContainers = 2).run()
    Test(boxCountPerStackableItem = 2, numberOfContainers = 2).run()
    Test(boxCountPerStackableItem = 2, numberOfContainers = 100).run()
}

class Test(
    private val boxCountPerStackableItem: Int,
    private val numberOfContainers: Int
) {
    fun run() {
        val containers: List<Container> = (1..numberOfContainers).map { containerIndex: Int ->
            Container.newBuilder()
                .withDescription(containerIndex.toString())
                .withSize(2352, 2394, 12031)
                .withEmptyWeight(4000)
                .withMaxLoadWeight(26480)
                .build()
        }

        val stackableItems: List<StackableItem> = listOf(
            createStackableItem(id = "1", width = 1200, height = 750, depth = 2280, weight = 285),
            createStackableItem(id = "2", width = 1200, height = 450, depth = 2280, weight = 155),
            createStackableItem(id = "3", width = 360, height = 360, depth = 570, weight = 20),
            createStackableItem(id = "4", width = 2250, height = 1200, depth = 2250, weight = 900),
            createStackableItem(id = "5", width = 1140, height = 750, depth = 1450, weight = 395),
            createStackableItem(id = "6", width = 1130, height = 1500, depth = 3100, weight = 800),
            createStackableItem(id = "7", width = 800, height = 490, depth = 1140, weight = 156),
            createStackableItem(id = "8", width = 800, height = 2100, depth = 1200, weight = 135),
            createStackableItem(id = "9", width = 1120, height = 1700, depth = 2120, weight = 160),
            createStackableItem(id = "10", width = 1200, height = 1050, depth = 2280, weight = 390)
        )

        val packager: LargestAreaFitFirstPackager = LargestAreaFitFirstPackager.newBuilder()
            .withContainers(containers)
            .build()

        val packedContainers = packager.packList(stackableItems, containers.size)

        println("Box count per stackable item: $boxCountPerStackableItem")
        println("Number of containers: $numberOfContainers")
        println("Packed containers: ${packedContainers?.size}")
        println("")
    }

    private fun createStackableItem(id: String, width: Int, height: Int, depth: Int, weight: Int): StackableItem {
        val box: Box = Box.newBuilder()
            .withId(id)
            .withSize(width, height, depth)
            .withWeight(weight)
            .withRotate3D()
            .build()

        return StackableItem(box, boxCountPerStackableItem)
    }
}
```

```
Box count per stackable item: 1
Number of containers: 2
Packed containers: 1

Box count per stackable item: 2
Number of containers: 2
Packed containers: null

Box count per stackable item: 2
Number of containers: 100
Packed containers: null
```

Thanks!

**Comment by skjolber** [comment]
Fixed in 2.0.5

**Comment by 4ybakut2004** [comment]
@skjolber, thanks! I tried version "2.0.5", now I received packed containers instead of `null` :smile: 

However, it seems that some of the boxes inside the containers intersect each other. For example, there are two boxes in the first container with the same position:

```
53. Position: 0x1700x10850
81. Position: 0x1700x10850
```

Here is the full source code:

```kotlin
import com.github.skjolber.packing.api.Box
import com.github.skjolber.packing.api.Container
import com.github.skjolber.packing.api.StackPlacement
import com.github.skjolber.packing.api.StackableItem
import com.github.skjolber.packing.packer.laff.LargestAreaFitFirstPackager

fun main(args: Array<String>) {
    Test(boxCountPerStackableItem = 1, numberOfContainers = 100).run()
}

class Test(
    private val boxCountPerStackableItem: Int,
    private val numberOfContainers: Int
) {
    fun run() {
        val containerEntity = ContainerEntity(
            width = 2352,
            height = 2394,
            depth = 12031,
            emptyWeight = 4000,
            maxLoadWeight = 26480
        )

        val boxEntities: List<BoxEntity> = listOf(
            BoxEntity(id = "1", width = 1200, height = 750, depth = 2280, weight = 285),
            BoxEntity(id = "2", width = 1200, height = 450, depth = 2280, weight = 155),
            BoxEntity(id = "3", width = 360, height = 360, depth = 570, weight = 20),
            BoxEntity(id = "4", width = 2250, height = 1200, depth = 2250, weight = 900),
            BoxEntity(id = "5", width = 1140, height = 750, depth = 1450, weight = 395),
            BoxEntity(id = "6", width = 1130, height = 1500, depth = 3100, weight = 800),
            BoxEntity(id = "7", width = 800, height = 490, depth = 1140, weight = 156),
            BoxEntity(id = "8", width = 800, height = 2100, depth = 1200, weight = 135),
            BoxEntity(id = "9", width = 1120, height = 1700, depth = 2120, weight = 160),
            BoxEntity(id = "10", width = 1200, height = 1050, depth = 2280, weight = 390),
            BoxEntity(id = "11", width = 900, height = 700, depth = 1200, weight = 114),
            BoxEntity(id = "12", width = 800, height = 1500, depth = 1200, weight = 506),
            BoxEntity(id = "13", width = 800, height = 1500, depth = 1200, weight = 513),
            BoxEntity(id = "14", width = 800, height = 1600, depth = 1200, weight = 592),
            BoxEntity(id = "15", width = 800, height = 1500, depth = 1200, weight = 318),
            BoxEntity(id = "16", width = 2300, height = 1700, depth = 3950, weight = 2260),
            BoxEntity(id = "17", width = 2300, height = 2250, depth = 4700, weight = 1080),
            BoxEntity(id = "22", width = 1100, height = 2200, depth = 1600, weight = 450),
            BoxEntity(id = "24", width = 2320, height = 1000, depth = 1650, weight = 140),
            BoxEntity(id = "25", width = 730, height = 750, depth = 1140, weight = 120),
            BoxEntity(id = "26", width = 400, height = 200, depth = 400, weight = 3),
            BoxEntity(id = "27", width = 1200, height = 750, depth = 2280, weight = 460),
            BoxEntity(id = "28", width = 1200, height = 750, depth = 2280, weight = 325),
            BoxEntity(id = "29", width = 1450, height = 750, depth = 1140, weight = 550),
            BoxEntity(id = "30", width = 1100, height = 2200, depth = 3200, weight = 550),
            BoxEntity(id = "31", width = 800, height = 800, depth = 1500, weight = 150),
            BoxEntity(id = "32", width = 1000, height = 1000, depth = 1900, weight = 350),
            BoxEntity(id = "33", width = 500, height = 2000, depth = 2300, weight = 350),
            BoxEntity(id = "34", width = 2300, height = 2300, depth = 4000, weight = 1710),
            BoxEntity(id = "35", width = 1400, height = 500, depth = 1400, weight = 650),
            BoxEntity(id = "36", width = 1600, height = 1200, depth = 4200, weight = 1500),
            BoxEntity(id = "37", width = 2300, height = 1800, depth = 4400, weight = 3235),
            BoxEntity(id = "38", width = 1200, height = 750, depth = 2280, weight = 370),
            BoxEntity(id = "40", width = 2300, height = 1200, depth = 2930, weight = 500),
            BoxEntity(id = "41", width = 2320, height = 1500, depth = 3300, weight = 1030),
            BoxEntity(id = "42", width = 2300, height = 1800, depth = 3300, weight = 1250),
            BoxEntity(id = "43", width = 2320, height = 1800, depth = 3300, weight = 625),
            BoxEntity(id = "44", width = 2300, height = 1200, depth = 2930, weight = 520),
            BoxEntity(id = "45", width = 2300, height = 2300, depth = 3300, weight = 1340),
            BoxEntity(id = "46", width = 1700, height = 1000, depth = 2300, weight = 400),
            BoxEntity(id = "47", width = 1700, height = 1000, depth = 2300, weight = 400),
            BoxEntity(id = "48", width = 1450, height = 750, depth = 1140, weight = 450),
            BoxEntity(id = "49", width = 2300, height = 500, depth = 3300, weight = 700),
            BoxEntity(id = "50", width = 1200, height = 750, depth = 2280, weight = 320),
            BoxEntity(id = "51", width = 1140, height = 750, depth = 1450, weight = 260),
            BoxEntity(id = "52", width = 730, height = 750, depth = 1140, weight = 105),
            BoxEntity(id = "53", width = 500, height = 800, depth = 2260, weight = 140),
            BoxEntity(id = "54", width = 1140, height = 450, depth = 1450, weight = 245),
            BoxEntity(id = "55", width = 1140, height = 450, depth = 1450, weight = 285),
            BoxEntity(id = "56", width = 1140, height = 450, depth = 1450, weight = 260),
            BoxEntity(id = "57", width = 1140, height = 750, depth = 1450, weight = 280),
            BoxEntity(id = "58", width = 1140, height = 750, depth = 1450, weight = 355),
            BoxEntity(id = "59", width = 1140, height = 450, depth = 1450, weight = 210),
            BoxEntity(id = "60", width = 1140, height = 750, depth = 1450, weight = 235),
            BoxEntity(id = "61", width = 1140, height = 450, depth = 1450, weight = 260),
            BoxEntity(id = "62", width = 650, height = 1400, depth = 850, weight = 20),
            BoxEntity(id = "63", width = 1150, height = 1700, depth = 2150, weight = 200),
            BoxEntity(id = "64", width = 2320, height = 2350, depth = 6900, weight = 1835),
            BoxEntity(id = "65", width = 370, height = 350, depth = 6200, weight = 110),
            BoxEntity(id = "66", width = 1050, height = 1800, depth = 6050, weight = 805),
            BoxEntity(id = "67", width = 850, height = 750, depth = 3400, weight = 290),
            BoxEntity(id = "68", width = 800, height = 1330, depth = 1220, weight = 70),
            BoxEntity(id = "69", width = 150, height = 250, depth = 6050, weight = 91),
            BoxEntity(id = "70", width = 850, height = 750, depth = 6200, weight = 605),
            BoxEntity(id = "71", width = 850, height = 900, depth = 4200, weight = 450),
            BoxEntity(id = "72", width = 1200, height = 620, depth = 1200, weight = 75),
            BoxEntity(id = "73", width = 750, height = 500, depth = 2600, weight = 170),
            BoxEntity(id = "74", width = 800, height = 500, depth = 1200, weight = 50),
            BoxEntity(id = "75", width = 370, height = 350, depth = 3400, weight = 180),
            BoxEntity(id = "76", width = 800, height = 770, depth = 1200, weight = 86),
            BoxEntity(id = "77", width = 1200, height = 750, depth = 2280, weight = 610),
            BoxEntity(id = "78", width = 800, height = 940, depth = 1200, weight = 100),
            BoxEntity(id = "79", width = 1600, height = 550, depth = 3200, weight = 700),
            BoxEntity(id = "80", width = 2320, height = 2200, depth = 2950, weight = 1000),
            BoxEntity(id = "81", width = 350, height = 370, depth = 2280, weight = 50),
            BoxEntity(id = "82", width = 1200, height = 250, depth = 2280, weight = 120),
            BoxEntity(id = "83", width = 1200, height = 450, depth = 2280, weight = 280),
            BoxEntity(id = "84", width = 2300, height = 1500, depth = 3300, weight = 800),
            BoxEntity(id = "85", width = 2300, height = 2000, depth = 3300, weight = 1000),
            BoxEntity(id = "87", width = 730, height = 750, depth = 1140, weight = 120),
            BoxEntity(id = "88", width = 2300, height = 2250, depth = 3300, weight = 1000),
            BoxEntity(id = "89", width = 1200, height = 900, depth = 2280, weight = 600),
            BoxEntity(id = "90", width = 2320, height = 2350, depth = 3950, weight = 1120),
            BoxEntity(id = "92", width = 2300, height = 2300, depth = 3300, weight = 820),
            BoxEntity(id = "93", width = 2300, height = 2250, depth = 2100, weight = 860),
            BoxEntity(id = "94", width = 2280, height = 2280, depth = 4800, weight = 2390),
            BoxEntity(id = "96", width = 1200, height = 950, depth = 2100, weight = 340),
            BoxEntity(id = "97", width = 2300, height = 2000, depth = 4200, weight = 1100),
            BoxEntity(id = "99", width = 2300, height = 2250, depth = 3300, weight = 1320),
            BoxEntity(id = "101", width = 600, height = 1250, depth = 800, weight = 60)
        )

        val containers: List<Container> = (1..numberOfContainers).map { containerIndex: Int ->
            Container.newBuilder()
                .withId(containerIndex.toString())
                .withDescription(containerIndex.toString())
                .withSize(containerEntity.width, containerEntity.height, containerEntity.depth)
                .withEmptyWeight(containerEntity.emptyWeight)
                .withMaxLoadWeight(containerEntity.maxLoadWeight)
                .build()
        }

        val stackableItems: List<StackableItem> = boxEntities.map { boxEntity: BoxEntity ->
            val box: Box = Box.newBuilder()
                .withId(boxEntity.id)
                .withSize(boxEntity.width, boxEntity.height, boxEntity.depth)
                .withWeight(boxEntity.weight)
                .withRotate3D()
                .build()

            StackableItem(box, boxCountPerStackableItem)
        }

        val packager: LargestAreaFitFirstPackager = LargestAreaFitFirstPackager.newBuilder()
            .withContainers(containers)
            .build()

        val packedContainers = packager.packList(stackableItems, containers.size)

        println("Box count per stackable item: $boxCountPerStackableItem")
        println("Number of containers: $numberOfContainers")

        if (packedContainers != null) {
            println("Packed containers: ${packedContainers.size}")

            packedContainers.forEach { container: Container ->
                println("  Container: ${container.description}")
                println("  Items:")

                container.stack.placements.forEach { placement: StackPlacement ->
                    val box = placement.stackable as Box
                    println("    ${box.description}. Position: ${placement.absoluteX}x${placement.absoluteY}x${placement.absoluteZ}")
                }
            }
        }

        println("")
    }
}

class ContainerEntity(
    val width: Int,
    val height: Int,
    val depth: Int,
    val emptyWeight: Int,
    val maxLoadWeight: Int
)

class BoxEntity(
    val id: String,
    val width: Int,
    val height: Int,
    val depth: Int,
    val weight: Int
)
```
